### DISCUSSION TITLE
Dynamic blacklist marker

### DISCUSSION THREAD

**Discussion by Manthhys** [Vdauphin/HeartsAndMinds#1277] [Question and answer] [answered]
Hi all,

I'd like to creat a dynamic blacklist area centered attached to a designed a FOB vehicule, but for that to work I'd need to know two things if aynone can help:
1- Is the city [init](https://github.com/Vdauphin/HeartsAndMinds/blob/master_daily/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/init.sqf) called at each server reboot or only once ?
2- Are the database vehicles instanciated before, or after the city init ?

Thanks in advance !

**Comment by Vdauphin** [comment, marked as answer]
Hello,

> I'd like to creat a dynamic blacklist area centered attached to a designed a FOB vehicule,

I don't really get what you are trying to achieved

> 1- Is the city init called at each server reboot or only once ?

It is called at each server reboot here:
https://github.com/Vdauphin/HeartsAndMinds/blob/036ffa1a73274858d5769abde5f6012ac21ea3b3/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/init_server.sqf#L1

> 2- Are the database vehicles instanciated before, or after the city init ?

The database is loaded after city init here:
https://github.com/Vdauphin/HeartsAndMinds/blob/3869216e1389c48a8830c18cfe79d03472b02587/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/init_server.sqf#L12

Cheers

**Reply by Manthhys** [reply]
Hey Vdauphin !
I'm was trying to see if this
if ((getMarkerPos "blacklistmarker") inArea [_position, 1800, 1800, 0, false]) exitWith {};
could detect my FOB vehicle after a restart to create a safezone around it.

Thanks for the answer ! So my solution wont work, maybe I could create a marker at the vehicle position when saving the databse or something ?

**Reply by Vdauphin** [reply]
> could detect my FOB vehicle after a restart to create a safezone around it.

No you can't

> So my solution wont work, maybe I could create a marker at the vehicle position when saving the databse or something ?

I would rather disable simulation of the trigger's city close to the vehicle
https://github.com/Vdauphin/HeartsAndMinds/blob/3869216e1389c48a8830c18cfe79d03472b02587/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L137

But I don't really understand. If you clean a city, only civilian will occupy the city. City can only switch again to occupied if an hideout is around.
Also a safe zone will not prevent enemy patrols. They will still use road around your vehicles.
Honestly, the best practice is just to move your vehicles/FOB somewhere safe so when you spawn the next day, you are not surrounded by enemies that move and patrol during the all day.

**Reply by Manthhys** [reply]
Hmm, okay so by default if I clean a city/location ennemies won't spawn inside it at restart even if there still hideouts on the map elsewhere ?

**Reply by Manthhys** [reply]
I must say I'm a bit confused as last time we tried to capture an area to establish a FOB we had ennemy spawning inside the base at restart eventhough we cleaned it before haha,

**Reply by Vdauphin** [reply]
Each city is generated the first time a player get around it. Then everything happening later will be store in the database. If you kill enemies or civilian, they will not spawn anymore.

If you kill all enemies in a city, this city will become free. But if there is a hideout around, troops will be send from the hideout to the city. If player doesn't defend it, they will be added to the city

You can see this behaviour easily by loading the mission in multiplayer and by enabling the debug in mission parameter. You will be allow to interact with yourself and turn on the debug graph and see AI position on map. You are also able to teleport yourself by clicking on map (player damage is disable in debug)

**Reply by Manthhys** [reply]
Alright thanks a lot !

**Reply by Vdauphin** [reply]
Have fun!
